stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala

```diff
@@ -267,9 +266,8 @@ private[stream] object Collect {
         } catch {
           case NonFatal(ex) =>
             decider(ex) match {
-              case Supervision.Stop    => failStage(ex)
-              case Supervision.Resume  => if (!hasBeenPulled(in)) pull(in)
-              case Supervision.Restart => if (!hasBeenPulled(in)) pull(in)
```

Comment:
can you explain the change? - does `pull(in)` automatically check `!hasBeenPulled(in)`?

Reply:
There is no need to check the `(!hasBeenPulled(in))` , because there is no additional `pull(in)`.  these method was copied from the old code, where, an exception maybe throw during the `preStart` but the upstream has been pulled. but here, as the `collect` operator never do some pulling or any other things which can throws after a `pull(in)`, so I think it would be better to just remove those duplicated check.

That's why I remove it.